graphql_compiler/schema_transformation/split_query.py

```diff
@@ -34,6 +46,478 @@ def __init__(self, query_ast):
+    branch that each cross-schema vertex field leads to will be made into its own separate query
+    AST. The parent and child property fields used in the stich will be added to the parent and
+    child ASTs, if not already present. @outpt directives will be added to these parent and
+    child property fields, if not already present. @filter directives will not be added to
+    child property fields in this step. This is because one may choose to rearrange and reroot
+    the tree of SubQueryNodes to achieve an execution order with better performance. @filter
+    directives should be added only once the tree's structure is fixed.
+
+    _split_query_ast_one_level_recursive will be called recursive on each intra-schema vertex
+    field.
+
+    Args:
+        query_node: SubQueryNode, whose list of child query connections may be modified to
+                    include new children
+        selections: List[Field], containing a number of property fields and vertex fields
+        type_info: TypeInfo, used to get information about the types of fields while traversing
+                   the query AST
+        edge_to_stitch_fields: Dict[Tuple(str, str), Tuple(str, str)], mapping
+                               (type name, vertex field name) to
+                               (source field name, sink field name) used in the @stitch directive
+                               for each cross schema edge
+        name_assigner: IntermediateOutNameAssigner, object used to generate and keep track of
+                       names of newly created @output directives
+
+    Returns:
+        List[Field], with which to replace the list of selections in the SelectionSet one level
+        above. All cross schema edges in the input list will be removed, and in their place,
+        property fields added or modified. If no changes were made, the exact input list object
+        will be returned
+    """
+    parent_type_name = type_info.get_parent_type().name
+
+    property_fields_map, vertex_fields = _split_selections_property_and_vertex(selections)
+    intra_schema_fields, cross_schema_fields = _split_vertex_fields_intra_and_cross_schema(
+        vertex_fields, parent_type_name, edge_to_stitch_fields
+    )
+
+    made_changes = False
+    if len(cross_schema_fields) > 0:
+        made_changes = True
+
+    # First, no changes will be made to property fields
+
+    # Second, process cross schema fields. This will modify our record of property fields, and
+    # create child SubQueryNodes attached to the input SubQueryNode
+    for cross_schema_field in cross_schema_fields:
+        type_info.enter(cross_schema_field)
+        child_type = type_info.get_type()
+        if child_type is not None:
+            child_type_name = strip_non_null_and_list_from_type(child_type).name
+        else:
+            raise AssertionError(
+                u'The query may be invalid against the schema, causing TypeInfo to lost track '
```

Comment:
typo: `causing TypeInfo to lost track` I think `lost -> lose`?

For bonus points, cross-schema field and maybe the query node at which this happened? Hitting this error would be quite scary and all the debug output we can provide would be helpful.